beesdoo_shift_swap/models/res_partner.py

```diff
@@ -1,11 +1,25 @@
-from odoo import _, api, models
+from datetime import datetime
+
+from odoo import _, api, fields, models
 from odoo.exceptions import UserError
 
 
+def get_validate_date(element):
```

Comment:
As @victor-champonnois suggests, it's probably better to have this function as a private static method.
Also, couldn't you overide this function in class `beesdoo.shift.exchange_request` to have it use `validate_date`?